src/protocol/ipa/mod.rs

```diff
@@ -380,27 +409,32 @@ where
         .collect::<Vec<_>>();
 
     let accumulated_credits = accumulate_credit(
-        sh_ctx.narrow(&Step::AccumulateCredit),
+        m_ctx.narrow(&Step::AccumulateCredit),
         &attribution_input_rows,
     )
     .await?;
 
     let user_capped_credits = credit_capping(
-        sh_ctx.narrow(&Step::PerformUserCapping),
+        m_ctx.narrow(&Step::PerformUserCapping),
         &accumulated_credits,
         per_user_credit_cap,
     )
     .await?;
 
-    aggregate_credit::<F, BK>(
-        sh_ctx.narrow(&Step::AggregateCredit),
+    //Validate before calling sort with downgraded context
+    // let user_capped_credits = malicious_validator.validate(user_capped_credits).await?;
+
+    malicious_aggregate_credit::<F, BK>(
```

Comment:
Returning Replicated<F> as output since there is compute after this and in aggregate_credit, last communication operation was sort